src/threading.c

```diff
@@ -613,6 +618,12 @@ void _jl_mutex_wait(jl_task_t *self, jl_mutex_t *lock, int safepoint)
         if (safepoint) {
             jl_gc_safepoint_(self->ptls);
         }
+        if (jl_running_under_rr(0)) {
+            uv_mutex_lock(&tls_lock);
+            if (jl_atomic_load_relaxed(&lock->owner))
+                uv_cond_wait(&cond, &tls_lock);
+            uv_mutex_unlock(&tls_lock);
+        }
```

Comment:
Why does this improve runtimes under `rr`?  It would be good to comment this so that others understand why this helps.